src/cleaning/cleaning.c

```diff
@@ -49,6 +49,156 @@ struct cleaning_policy_ops cleaning_policy_ops[ocf_cleaning_max] = {
 	},
 };
 
+void ocf_cleaning_setup(ocf_cache_t cache, ocf_cleaning_t policy)
+{
+	ENV_BUG_ON(policy >= ocf_cleaning_max);
+
+	if (!cleaning_policy_ops[policy].setup)
+		return;
+
+	cleaning_policy_ops[policy].setup(cache);
+}
+
+int ocf_cleaning_initialize(ocf_cache_t cache, ocf_cleaning_t policy,
+		int init_metadata)
+{
+	ENV_BUG_ON(policy >= ocf_cleaning_max);
+
+	if (!cleaning_policy_ops[policy].initialize)
+		return 0;
+
+	return cleaning_policy_ops[policy].initialize(cache, init_metadata);
+}
+
+void ocf_cleaning_deinitialize(ocf_cache_t cache, ocf_cleaning_t policy)
+{
+	ENV_BUG_ON(policy >= ocf_cleaning_max);
+
+	if (!cleaning_policy_ops[policy].deinitialize)
+		return;
+
+	cleaning_policy_ops[policy].deinitialize(cache);
+}
```

Comment:
I've left `policy` param in init function because rollback procedure would become a bit tricky. In the current implementation we don't have to change `cache->conf_meta->cleaning_policy_typ` value until new cleaning policy is initialized.